From 5f37a6e943cb84fa91c7106b235cb775240a4e06 Mon Sep 17 00:00:00 2001
From: Murillo Arantes <murillo.arantes@windriver.com>
Date: Wed, 15 Apr 2026 11:50:27 -0300
Subject: [PATCH] STX: Fixed default libvirtd systemd service for pmon

This patch merges 0005-STX-Fixed-default-libvirtd-systemd-service
and 0006-STX-Adding-systemd-dependencies-against-pmon.service patches,
as both modify the same file for pmon related changes.

- 'Restart=no' (pmond will monitor the process)
- 'PIDFile=/var/run/libvirtd.pid' (pmond uses the pidfile to monitor the
  processes)
- 'Before=pmon.service' (libvirtd must start before pmond)

Signed-off-by: Ludovic Beliveau <ludovic.beliveau@windriver.com>
Signed-off-by: Don Penney <don.penney@windriver.com>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
[ Ported to libvirt v11.3.0-3+deb13u2; merged pmon patches ]
Signed-off-by: Murillo Arantes <murillo.arantes@windriver.com>
---
 src/remote/libvirtd.service.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/remote/libvirtd.service.in b/src/remote/libvirtd.service.in
index b0a062e88..be5c56a68 100644
--- a/src/remote/libvirtd.service.in
+++ b/src/remote/libvirtd.service.in
@@ -22,6 +22,7 @@ After=iscsid.service
 After=apparmor.service
 After=remote-fs.target
 After=systemd-machined.service
+Before=pmon.service
 After=xencommons.service
 Conflicts=xendomains.service
 
@@ -32,7 +33,8 @@ EnvironmentFile=-@initconfdir@/libvirtd
 ExecStart=@sbindir@/libvirtd $LIBVIRTD_ARGS
 ExecReload=/bin/kill -HUP $MAINPID
 KillMode=process
-Restart=on-failure
+Restart=no
+PIDFile=/var/run/libvirtd.pid
 
 [Install]
 WantedBy=multi-user.target
-- 
2.34.1

